### PR TITLE
Adds eslint to vuepress-theme-odyssey

### PR DESCRIPTION
**Pull request by arnoldsandoval-okta**
Adds `eslint` and updates code style using [eslint-config-vuepress](https://www.npmjs.com/package/eslint-config-vuepress)

### CODE REVIEW

**Comment by arnoldsandoval-okta** [comment]
Note: The scripts block in `packages/vuepress-theme-odyssey` was also changed. I accidentally checked this in as part of #728 

https://github.com/okta/odyssey/blob/1997c567014f2e571ac3f655b08b5db59d27e009/packages/vuepress-theme-odyssey/package.json#L8-L11

**Review by santhoshbalakrishnan — APPROVED** [review]
🍰

File: package.json
Comment (jmelberg-okta) [inline package.json:19]: Let's add a `lint` script to the `scripts` object here so we can actually invoke this change! 😄

**Review by jmelberg-okta — CHANGES_REQUESTED** [review]

File: package.json
Comment (arnoldsandoval-okta) [reply package.json:21]: I forgot to push it up, was on my box. It is now fixed!

File: package.json
Comment (arnoldsandoval-okta) [inline package.json:17]: I feel a little weird about how I chose to specify the file extension in the glob instead of the eslint flag for extension, but the latter wasn't working as expected. This does!

Comment (jmelberg-okta) [reply package.json:17]: I'd assume that the plugin should provide the necessary overrides to run this on `js` and `vue` extensions. We should be able to do the following to simply this call:

1. Create an `.eslintignore` file that has `node_modules` and `dist` (I assume this is the VuePress build output) added to it
2. Update this script to run on all packages in this mono-repo:
   ```diff
   -"eslint '**/*.@(js|vue)' --ignore-pattern node_modules/ && stylelint ./*.scss"
   +"eslint . && stylelint ./*.scss"
   ```

File: packages/vuepress-theme-odyssey/.eslintrc.js
Comment (jmelberg-okta) [inline packages/vuepress-theme-odyssey/.eslintrc.js:3]: Since we aren't providing any rule overrides - we can remove this object:

```suggestion
```

File: packages/vuepress-theme-odyssey/package.json
Comment (jmelberg-okta) [inline packages/vuepress-theme-odyssey/package.json:12]: Since this is likely going to be used on the docs site as well, I'd suggest moving this to the root dependency list.

File: packages/vuepress-theme-odyssey/styles/base/_fonts.scss
Comment (jmelberg-okta) [inline packages/vuepress-theme-odyssey/styles/base/_fonts.scss:1]: Please remove this file!

File: package.json
Comment (arnoldsandoval-okta) [reply package.json:17]: I had tried the approach you are outlining here earlier and got the following result:

image

Note: there is an error in the `.vue` file above.

When I add the  `--ext` flag to include `.vue` files it works fine

image

Pushing it up as is, let me know if you think this requires further changes.

File: packages/vuepress-theme-odyssey/package.json
Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/package.json:12]: Done

File: packages/vuepress-theme-odyssey/styles/base/_fonts.scss
Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/styles/base/_fonts.scss:1]: Weird, not sure how this got here. Removed.

File: package.json
Comment (jmelberg-okta) [reply package.json]: This is likely because `lint` is being run from the package instead of the root - but I'm OK with this for now.

**Review by jmelberg-okta — APPROVED** [review]
Looks good!

Please hold off on merging this into the `vuepress-theme-odyssey` branch until #728 is merged into `develop`, otherwise we're back at where we started. Then, let's rebase this against `develop` and squash + merge.

**Comment by jmelberg-okta** [comment]
@arnoldsandoval-okta - Looks like we also don't have any `yarn.lock` file updates. Please push a new commit with `yarn install` ran on the root of this project to install the new dependencies!

**Pull request merged**
